modules/imgproc/src/smooth.dispatch.cpp

```diff
@@ -747,6 +768,26 @@ void GaussianBlur(InputArray _src, OutputArray _dst, Size ksize,
                          ofs.x, ofs.y, wsz.width - src2.cols - ofs.x,  wsz.height - src2.rows - ofs.y, ksize.width, borderType&~BORDER_ISOLATED);
             }
 
+            if (allow_approximations)
```

Comment:
Maybe it would be better to add this condition (negated) conditions on lines 657 and 726? Currently it looks like this:
```.cpp
if (depth==CV_8U && <something>)
{
     // ... more conditions
    // trying OCL - ocl_sepFilter2D_BitExact
    // trying HAL - gaussianBlurBinomial
    if (allow_approximations)
    {
        // trying IPP - ipp_GaussianBlur
        // trying HAL - gaussianBlur
        // trying OVX
    }
    // call GaussianBlurFixedPoint
    return;
}
if (depth==CV_16U && <something>)
{
    // similar scheme
    return;
}
// try OCL - sepFilter2D
// try HAL - gaussianBlur
// try OVX
// try IPP - ipp_GaussianBlur
// call sepFilter2D
```
Maybe we could just add `!allow_approximations` in both bit-exact branches and fall to the last non-bit-exact block?

---

Also, why is IPP called before HAL in new blocks? Usually we prefer HAL to anything else.

Reply:
Fixed HAL-IPP order. In case if "faster" branch does not work or not available we need to fallback to bit-exact implementation.